src/Components/App/App.js

```diff
@@ -14,13 +16,37 @@ class App extends Component {
       moviesAnswers:[],
       podcastsAnswers:[],
       boardGamesAnswers:[],
-      cardGamesAnswers:[]
+      cardGamesAnswers:[],
+      movies: [],
+      music: [],
+      podcasts: [],
+      boardGames: [],
+      cardGames: []
     }
   }
 
-  findSetOfQuestions = () => {
-  
-    // this.state
+  getActivityData = async (event) => {
+    let promise;
+    try{
+      if(event.target.id === 'movies' ) {
+          promise = await getAllMovies();
+      }
+      if(event.target.id === 'boardGames' ) {
+        promise = await getAllBoardGames();
+      }
+      if(event.target.id === 'cardGames' ) {
+        promise = await getAllCardGames();
+      }
+      if(event.target.id === 'music' ) {
+          promise = await getAllMusic();
+      }
+      if(event.target.id === 'podcasts' ) {
+        promise = await getAllPodcasts();
+      }
+      this.setState({ [event.target.id]: promise})
+      } catch (error){
+        console.log(error)
+      }
```

Comment:
Keeping in mind that we don't want to make all network requests at the same time for all the information when the page loads to slow our application, I decided to make the request one at a time using the navigation bar`. I don't think that the scope of the project is really needed but I can imagine that when working on larger projects this making individual requests would be a better idea. Any other thoughts?

Reply:
I agree.  I like the idea of only fetching the data when the data is needed.  Where is `getActivityData()` being invoked?  I'm assuming the `event` that's passed as an argument is the user clicking on an activity in the `Form`, but just want to clarify.

Reply:
I think it's a great Idea, I don't see any reason why would need that information at any other scope than this. I think if we needed that information to persist or be available in multiple views would be the only reason this wouldn't work